### PR TITLE
feat: (homepage): Bring back username visibility toggle

### PR DESCRIPTION
**Pull request by memoyil**

### CODE REVIEW

**Comment by Woody810** [comment]
There is no toggle in the profile page

File: src/views/TradingCompetition/components/RegisterModal/RegisterWithProfile.tsx [hunk 1]
```diff
@@ -44,7 +46,7 @@ const RegisterWithProfile: React.FC<React.PropsWithChildren<CompetitionProps>> =
 
   return (
     <>
-      <Heading scale="md" mb="24px">{`@${profile.username}`}</Heading>
+      <Heading scale="md" mb="24px">{`@${userUsernameVisibility ? profile.username : '🐰🐰🐰🐰🐰🐰'}`}</Heading>
```
Comment (chef-jojo) [inline src/views/TradingCompetition/components/RegisterModal/RegisterWithProfile.tsx:49]: i think we can have a hook to merging profile + visibility and the output

**Comment by chef-jojo** [comment]
Should we show username by default?

**Comment by chef-jojo** [comment]
> > Should we show username by default?
> 
> The default value of that parameter is false, if want to make it default true we should add migrate in redux persist or create a new parameter

yeah it's tricky. That's why we have some flags are `disabledXXXX` instead

**Review by chef-jojo — DISMISSED** [review]

**Review by chef-jojo — DISMISSED** [review]

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**